elastic/logstash
# ECS on by default for Logstash 8, again

## Release notes

[breaking] The `pipeline.ecs_compatibility` setting now defaults to `v8`, causing all plugins that implement an ECS compatibility mode to default to operating with `ecs_compatibility => v8`.

## What does this PR do?

Effectively reverts #13080, making the _default_ ECS compatibility mode `v8` for Logstash 8, and adds INFO-level logging for each pipeline to indicate in which mode its plugins will be run.

## Why is it important/What is the impact to the user?

This the final step of a large and ongoing effort to provide better support for the Elastic Common Schema, detailed in #11623. In #13080 we had back-tracked making the breaking change out of an abundance of caution, but have since decided to move forward with it.

## Checklist


## Author's Checklist

There are a number of plugins that ship with Logstash and have ECS modes, that do NOT yet have modes for v8. Because [the scope of changes for v1->v8 is so small](https://github.com/elastic/ecs/issues/839), I _think_ that all of these can provide a v8 implementation that simply aliases the existing v1 implementation.

- [x] geoip filter: https://github.com/logstash-plugins/logstash-filter-geoip/pull/193
- [x] grok filter: https://github.com/logstash-plugins/logstash-filter-grok/pull/175 (alias, but log warning; need to validate against v1->v8 scope changes in finer detail)
- [x] clone filter: https://github.com/logstash-plugins/logstash-filter-clone/pull/27
- [x] syslog_pri filter: https://github.com/logstash-plugins/logstash-filter-syslog_pri/pull/10
- [x] useragent filter: https://github.com/logstash-plugins/logstash-filter-useragent/pull/76
- [x] syslog input: https://github.com/logstash-plugins/logstash-input-syslog/pull/68

## How to test this PR locally

~~~ sh
bin/logstash -e ''
~~~

 - Observe an info-level log indicating that the pipeline is being run in ECS-compatibility mode.
 - Observe no deprecation log from the stdin input indicating that the default ecs_compatibility may change
 - Process one event, observe that the structure is ECS-compliant (`[host][hostname]`)

## Related issues

Resolves #11623 
Reverts #13080
